pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -544,37 +547,72 @@ func (qjm *XController) GetAggregatedResourcesPerGenericItem(cqj *arbv1.AppWrapp
 	// Get all pods and related resources
 	for _, genericItem := range cqj.Spec.AggrResources.GenericItems {
 		itemsList, _ := genericresource.GetListOfPodResourcesFromOneGenericItem(&genericItem)
-		for i :=0; i < len(itemsList); i++ {
+		for i := 0; i < len(itemsList); i++ {
 			retVal = append(retVal, itemsList[i])
 		}
 	}
 
 	return retVal
 }
+
+func (qjm *XController) GetAllObjectsOwned(cqj *arbv1.AppWrapper) {
+
+	// Get all pods and related resources
+	countCompletedItems := 0
+	countCompletionRequired := 0
+	ignoredItems := 0
+	for _, genericItem := range cqj.Spec.AggrResources.GenericItems {
+		objectName := genericItem.GenericTemplate
+		var unstruct unstructured.Unstructured
+		unstruct.Object = make(map[string]interface{})
+		var blob interface{}
+		if err := jsons.Unmarshal(objectName.Raw, &blob); err != nil {
+			klog.Errorf("Error unmarshalling, err=%#v", err)
+		}
+		if len(genericItem.CompletionStatus) > 0 {
+			countCompletionRequired = countCompletionRequired + 1
+		}
+		status := qjm.genericresources.GetGenericItemKindStatus(&genericItem, cqj.Namespace)
+		if status == "true" {
+			countCompletedItems = countCompletedItems + 1
+		}
+		//ignore service and podgroups since these items do not have status
+		if status == "ignore" {
+			ignoredItems = ignoredItems + 1
+		}
```

Comment:
See comment above regarding logic checking of new `completionstatus`

Reply:
If we start ignoring items without `completionstatus` I think none of the ray jobs will complete with the current design.

Reply:
ok, I think the above is not entirely true. please consider the case, if AW has job and deployment as two items and only job is tracked, we still set the status as `RunningHoldCompletion`. So non-compute items are treated differently than the compute items.